pkg/controller/syncset/syncset_controller_test.go

```diff
@@ -304,3 +306,61 @@ func testHashCompute(obj interface{}) (string, error) {
 		return "unknown", nil
 	}
 }
+
+func TestCDFirstSyncSetsSuccessTimestamp(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+
+	now := metav1.Now()
+
+	cd := testClusterDeployment()
+	cdInstalledTimestamp := metav1.NewTime(now.Add(-time.Hour * 1))
+	cd.Status.InstalledTimestamp = &cdInstalledTimestamp
+
+	ss := testMatchingSyncSet
+	si := testSyncSetInstanceForSyncSet
+
+	syncSets := []runtime.Object{
+		ss("a"), ss("b"), ss("c"),
+	}
+
+	// truncate time to the second to match rfc3339
+	firstAppliedTimes := []metav1.Time{
+		metav1.NewTime(now.Add(-time.Minute * 15).Truncate(time.Second)),
+		metav1.NewTime(now.Add(-time.Minute * 10).Truncate(time.Second)),
+		metav1.NewTime(now.Add(-time.Minute * 5).Truncate(time.Second)),
+	}
+	syncSetInstances := []runtime.Object{
+		si("a"), si("b"), si("c"),
+	}
+	for i, ssi := range syncSetInstances {
+		ssi.(*hivev1.SyncSetInstance).Status.FirstSuccessTimestamp = &firstAppliedTimes[i]
+	}
+
+	objs := append(syncSets, syncSetInstances...)
+	objs = append(objs, cd)
+	fakeClient := fake.NewFakeClient(objs...)
+	rss := &ReconcileSyncSet{
+		Client:      fakeClient,
+		scheme:      scheme.Scheme,
+		logger:      log.WithField("controller", "syncset"),
+		computeHash: testHashCompute,
+	}
+	_, err := rss.Reconcile(reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      testName,
+			Namespace: testNamespace,
+		},
+	})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return
+	}
+
+	uCD := &hivev1.ClusterDeployment{}
+	err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: testName, Namespace: testNamespace}, uCD)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return
+	}
+	assert.Equal(t, *uCD.Status.FirstSyncSetsSuccessTimestamp, *syncSetInstances[2].(*hivev1.SyncSetInstance).Status.FirstSuccessTimestamp)
```

Comment:
The expected value should be first. Also, it will read better if you store the expected value in a variable rather than reading from a field in a slice.
```suggestion
	assert.Equal(t, expectedFirstSuccessTime, *uCD.Status.FirstSyncSetsSuccessTimestamp, "unexpected first success timestamp")
```